src/components/ExportDataModal.js

```diff
@@ -0,0 +1,228 @@
+      }),
+      formLabel('Entries selected'),
+      ...Utils.cond(
+        [moreToDelete, () => this.displayEntities(additionalDeletions, runningSubmissionsCount, true)],
+        [(hardConflicts.length !== 0), () => this.displayEntities(hardConflicts, runningSubmissionsCount, true)],
+        [(softConflicts.length !== 0), () => this.displayEntities(softConflicts, runningSubmissionsCount, true)],
+        () => this.displayEntities(selectedEntities, runningSubmissionsCount, false)
+      ),
+      div({
+        style: { ...warningStyle, textAlign: 'right', marginTop: (hardConflicts.length !==0) ? '1rem' : undefined }
+      }, [`${selectedEntities.length} data entries to be copied.`]),
+      copying && spinnerOverlay,
+      error && h(ErrorView, { error, collapses: false })
+    ])
+  }
+
+  renderPostCopy() {
+    const { onDismiss } = this.props
+    const selectedWorkspace = this.getSelectedWorkspace().workspace
+
+    return h(Modal, {
+      title: 'Copy to Workspace',
+      onDismiss,
+      cancelText: 'Stay Here',
+      okButton: buttonPrimary({
+        onClick: () => {
+          Nav.goToPath('workspace-data', {
+            namespace: selectedWorkspace.namespace,
+            name: selectedWorkspace.name
+          })
+        }
+      }, ['Go to copied data'])
+    }, [
+      'Successfully copied data to ',
+      b([selectedWorkspace.name]),
+      '. Do you want to view the copied data?'
+    ])
+  }
+
+  displayEntities(entities, runningSubmissionsCount, showType) {
+    return _.map(([i, entity]) => div({
+      style: {
+        borderTop: (i === 0 && runningSubmissionsCount === 0) ? undefined : Style.standardLine,
+        padding: '0.6rem 1.25rem', margin: '0 -1.25rem'
+      }
+    }, showType ? `${entity.entityName} (${entity.entityType})` : entity),
+    Utils.toIndexPairs(entities))
+  }
+
+  async doOverride() {
+    const { onDismiss, ajax: { Workspaces } } = this.props
+    const { selectedEntityType, additionalDeletions, hardConflicts } = this.state
+    const selectedWorkspace = this.getSelectedWorkspace().workspace
+    const entitiesToDelete = _.concat(hardConflicts, additionalDeletions)
+    try {
+      await Workspaces.workspace(selectedWorkspace.namespace, selectedWorkspace.name).deleteEntities(entitiesToDelete)
+      this.setState({ hardConflicts: [], additionalDeletions: [] })
+    } catch (error) {
+      switch (error.status) {
+        case 409:
+          this.setState({
+            additionalDeletions: _.filter(entity => entity.entityType !== selectedEntityType,
+              await error.json()), copying: false
+          }) //handles dangling references when deleting entities
+          return
+        default:
+          reportError('Error deleting data entries', error)
+          onDismiss()
+      }
+    }
+    this.copy()
```

Comment:
The logic is well de-duplicated, but right now `copy` and `doOverride` are co-recursive, and rely on the state change to terminate. Even if this works now, it's risky because there's no guarantee that the results of a `setState` will be visible during the current tick. What about combining everything into a single function with a flow like:
```js
async copy() {
  if (hardConflicts.length) {
    // delete entries
  }
  // copy entries
}
```
with a couple bail-out points for stopping early to report conflicts.